build.rs

```diff
@@ -137,20 +137,19 @@ fn build_v8(is_asan: bool) {
   if need_gn_ninja_download() {
     download_ninja_gn_binaries();
   }
-
+  let target_os = env::var("CARGO_CFG_TARGET_OS").unwrap();
```

Comment:
> Can you add a few lines here with a note explaining why `cfg!` isn't appropriate for future contributors to this file? The commentary from the PR description will work.

Of course! Actually #[cfg(...)] attributes don't work as expected from build.rs -- they refer to the configuration of the host system which the build.rs script will be running on. In short, `cfg!(target_<os/arch>)` is actually the `host os/arch` instead of target os/arch while cross compiling. Instead, [Environment viariables](https://doc.rust-lang.org/cargo/reference/environment-variables.html#environment-variables-cargo-sets-for-build-scripts) is the officially approach to get the target os/arch in build.rs.

Reply:
Maybe some parts of the original code that use the cfg!(target_<os/arch>) macro are used to obtain the host <os/arch>, but I think using this ambiguous syntax is not conducive to long-term maintenance. It's highly recommended to use the `HOST` environment variable to get the host triplet or use cfg!(target_<os/arch>) with explicit annotation